api/screening/screeningRoute.js

```diff
@@ -22,69 +25,109 @@ router
   })
   // @route POST api/screenings
   // @description Create a screening
-  // @access admin
-  .post(authMiddleware, adminMiddleware, async (req, res) => {
-    // eslint-disable-next-line object-curly-newline
-    const { movieId, cinemaHallId, price, startDate } = req.body;
-    try {
-      const movie = await Movie.findById(movieId);
-      const cinemaHall = await CinemaHall.findById(cinemaHallId);
-      if (movie === undefined) {
-        res.status(400).json({
-          error: `Cannot find movie with id: ${req.params.id}'`,
-        });
-        return;
+  // @access Admin
+  .post(
+    authMiddleware,
+    adminMiddleware,
+    check('movieId', 'Please enter movie id').trim().notEmpty(),
+    check('cinemaHallId', 'Please enter cinema hall id')
+      .trim()
+      .notEmpty(),
+    check('price.normal', 'Please enter price').notEmpty().isFloat(),
+    check('price.reduced', 'Please enter price').notEmpty().isFloat(),
+    check('startDate', 'Start date needs to be a valid date')
+      .trim()
+      .isISO8601()
+      .toDate(),
+
+    async (req, res) => {
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
       }
-      if (cinemaHall === undefined) {
-        res.status(400).json({
-          error: `Cannot find cinemaHall with id: ${req.params.id}'`,
+      // eslint-disable-next-line object-curly-newline
+      const { movieId, cinemaHallId, price, startDate } = req.body;
+      try {
+        const movie = await Movie.findById(movieId);
+        const cinemaHall = await CinemaHall.findById(cinemaHallId);
+        if (movie === undefined) {
+          return res.status(400).json({
+            error: `Cannot find movie with id: ${req.params.id}'`,
+          });
+        }
+        if (cinemaHall === undefined) {
+          return res.status(400).json({
+            error: `Cannot find cinemaHall with id: ${req.params.id}'`,
+          });
+        }
+        const conflictingScreenings = await Screening.find({
+          $or: [
+            {
+              startDate: {
+                $gte: startDate,
+                $lte: modifyDate(startDate, movie.duration + 10),
+              },
+              cinemaHall: cinemaHallId,
+            },
+            {
+              endDate: {
+                $gte: modifyDate(startDate, -10),
+                $lte: modifyDate(startDate, movie.duration),
+              },
+              cinemaHall: cinemaHallId,
```

Comment:
Great you're checking for conflicted screenings. A check we should add is a check with the cinemas opening hours probably.

Reply:
I'll work on that in a separate PR.